Core/src/Geometry/LayerCreator.cpp

```diff
@@ -55,8 +56,7 @@ Acts::MutableLayerPtr Acts::LayerCreator::cylinderLayer(
     std::vector<std::shared_ptr<const Surface>> surfaces, std::size_t binsPhi,
     std::size_t binsZ, std::optional<ProtoLayer> _protoLayer,
     const Transform3& transform, std::unique_ptr<ApproachDescriptor> ad) const {
-  ProtoLayer protoLayer =
-      _protoLayer ? *_protoLayer : ProtoLayer(gctx, surfaces);
+  ProtoLayer protoLayer = _protoLayer.value_or(ProtoLayer(gctx, surfaces));
```

Comment:
I don't think there are functionally equivalent. In the ternary case that we had before, the constructor of `ProtoLayer` is only called in the else branch, whereas in this case the `ProtoLayer` constructor is **always** called.

That's not necessarily the end of the world, but it's not really what we want here.